addons/mail/models/discuss/discuss_channel.py

```diff
@@ -541,7 +555,10 @@ def _subscribe_users_automatically_get_members(self):
             )
 
     def action_unfollow(self):
-        self._action_unfollow(self.env.user.partner_id)
+        if self.channel_type in self._types_allowing_unfollow():
+            self._action_unfollow(self.env.user.partner_id)
+        else:
+            self.self_member_id.unpin_dt = fields.Datetime.now()
```

Comment:
Might be too heavy for the current PR, but I would expect to call the proper method. it looks odd to call `action_unfollow`, only to decide we won't unfollow but unpin instead.